internal/controller/isbservicerollout/isbservicerollout_controller.go

```diff
@@ -332,8 +327,28 @@ func (r *ISBServiceRolloutReconciler) processExistingISBService(ctx context.Cont
 		}
 	}
 
+	// don't risk out-of-date cache while performing PPND or Progressive strategy - get
+	// the most current version of the isbsvc just in case
+	if inProgressStrategy != apiv1.UpgradeStrategyNoOp {
+		existingISBServiceDef, err = kubernetes.GetLiveResource(ctx, newISBServiceDef, "interstepbufferservices")
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				numaLogger.WithValues("isbsvcDefinition", *newISBServiceDef).Warn("InterstepBufferService not found.")
+			} else {
+				return false, fmt.Errorf("error getting InterstepBufferService for status processing: %v", err)
+			}
+		}
+		newISBServiceDef = r.merge(existingISBServiceDef, newISBServiceDef)
+	}
+
 	switch inProgressStrategy {
 	case apiv1.UpgradeStrategyPPND:
+
+		_, isbServiceIsUpdating, err := r.isISBServiceUpdating(ctx, isbServiceRollout, existingISBServiceDef, true)
+		if err != nil {
+			return false, fmt.Errorf("error determining if ISBService is updating: %v", err)
+		}
```

Comment:
by moving the call down here, we can use our live version of `existingISBServiceDef`